lang/codegen/src/accessors.rs

```diff
@@ -0,0 +1,169 @@
+use proc_macro2::TokenStream;
+use quote::{
+    format_ident,
+    quote,
+    quote_spanned,
+};
+use syn::{
+    spanned::Spanned,
+    Data,
+    DataStruct,
+    Field,
+    Fields,
+};
+
+pub fn accessors(attrs: TokenStream, s: synstructure::Structure) -> TokenStream {
+    let trait_ident = attrs.clone();
+
+    let struct_ident = s.ast().ident.clone();
+
+    let item = match s.ast().data.clone() {
+        Data::Struct(struct_item) => generate_struct(&s, struct_item),
+        _ => panic!("Only structs are supported"),
+    };
+
+    let fields: Vec<_> = extract_get_fields(s.clone());
+
+    let trait_get_messages = fields.iter().map(|field| {
+        let field_ident = field.ident.clone().unwrap();
+        let method_ident = format_ident!("get_{}", field_ident);
+        let field_type = field.ty.clone();
+        let span = field.span();
+
+        quote_spanned! {span =>
+            #[ink(message)]
+            fn #method_ident(&self) -> #field_type;
+        }
+    });
+
+    let get_impls = fields.iter().map(|field| {
```

Comment:
if the field is Mapping here we will get error, same with setters

Reply:
we can do `get_mapping(&self, key: K) -> Option<V>` and `set_mapping(&mut self, key: K, value: V)`

Reply:
yeah, but the idea was to work with simple getters/setters, not with mappings, I'm actually not sure how the macro should determine whether the field is mapping or not

Reply:
it will be a bit complicated currently, lets add a TODO and mention in docs that we should not do something like that for now

Reply:
it is already added in documentation, that accessors work only for fields that implement scale::Encode and scale::Decode, so it's okay